modules/gdscript/gdscript_compiler.cpp

```diff
@@ -165,7 +165,30 @@ GDScriptDataType GDScriptCompiler::_gdtype_from_datatype(const GDScriptParser::D
 						}
 					}
 					if (result.script_type_ref.is_null()) {
-						result.script_type_ref = GDScriptCache::get_shallow_script(p_datatype.script_path, main_script->path);
+						if (classes.size() > 1) {
+							auto find_subclass_script = [&classes](Ref<GDScript> lp_script) -> Ref<GDScript> {
```

Comment:
We avoid the use of `auto` and lambdas in Godot code unless strictly needed. This could be a helper method instead.

Reply:
👍